store/tikv/2pc.go

```diff
@@ -411,6 +299,20 @@ func (c *twoPhaseCommitter) primary() []byte {
 	return c.primaryKey
 }
 
+// asyncSecondaries returns all keys that must be checked in the recovery phase of an async commit.
+func (c *twoPhaseCommitter) asyncSecondaries() [][]byte {
+	secondaries := make([][]byte, len(c.mutations.keys)-1)
```

Comment:
maybe we need use `make([][]byte, 0, len(c.mutations.keys)-1)` in here and `secondaries = append(secondaries, k)` in line 311?

for current, if we have mutations like `{key: pk1, op: put}, {key: k1, op: put}, {key: k2, op:checkNotExists}, {key: k2, op: put}` (first mutation is pk), we will get secondaries:

`[][]byte{{'k1'}, {'k2'}, nil}`

but maybe we real need is :

`[][]byte{{'k1'}, {'k2'}}` ~?

Reply:
rest LGTM